File: LeetCode/66.Plus_One/Mario/66.Plus_One.java
class Solution {
    public int[] plusOne(int[] digits) {
        int plus = 0;
        for (int i = digits.length - 1; i > -1; i--) {
            int val = digits[i] + plus + (i == digits.length - 1 ? 1 : 0);
            if (val > 9) {
                plus = 1;
                val %= 10;
            } else {
                plus = 0;
            }
            digits[i] = val;
        }
        if (plus == 1) {
            int[] res = new int[digits.length + 1];
            res[0] = plus;
            for (int i = 1; i < res.length; i++) {
                res[i] = digits[i - 1];
            }
            return res;
        }
        return digits;
    }
}

File: LeetCode/66.Plus_One/Mario/66.Plus_One.py
#!/usr/bin/env python3


class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 1
        for i in range(len(digits) - 1, -1, -1):
            val = digits[i] + carry
            digits[i] = val % 10
            carry = val // 10

            if carry == 0:
                break

        if carry == 1:
            digits.insert(0, 1)

        return digits
